qds_sdk/cloud/aws_cloud.py

```diff
@@ -0,0 +1,166 @@
+    '''
+
+    def __init__(self):
+        self.compute_config = {}
+        self.location = {}
+        self.network_config = {}
+        self.storage_config = {}
+
+    def set_cloud_config(self,
+                         compute_access_key=None,
+                         compute_secret_key=None,
+                         use_account_compute_creds=None,
+                         aws_region=None,
+                         aws_availability_zone=None,
+                         role_instance_profile=None,
+                         vpc_id=None,
+                         subnet_id=None,
+                         persistent_security_groups=None,
+                         bastion_node_public_dns=None,
+                         master_elastic_ip=None):
+        '''
+
+        Args:
+            compute_access_key: The access key for customer's aws account. This
+                is required for creating the cluster.
+
+            compute_secret_key: The secret access key for customer's aws
+                account. This is required for creating the cluster.
+
+            use_account_compute_creds: Set it to true to use the account’s compute
+                credentials for all clusters of the account.The default value is false
+
+            aws_region: The AWS region in which the cluster is created. The default value is, us-east-1.
+                Valid values are, us-east-1, us-west-1, us-west-2, eu-west-1, sa-east1, ap-southeast-1,
+                and ap-northeast-1.
+                Doc: http://docs.qubole.com/en/latest/rest-api/cluster_api/create-new-cluster.html#ec2-settings
+
+            aws_availability_zone: The preferred availability zone in which the cluster must be created. The default value is Any.
+
+            role_instance_profile: IAM Role instance profile to attach on cluster
+
+            vpc_id: The ID of the vpc in which the cluster is created.
+                In this vpc, the enableDnsHostnames parameter must be set to true.
+
+            subnet_id: The ID of the subnet in which the cluster is created. This subnet must belong to the
+                above VPC and it can be a public/private subnet
+
+            persistent_security_groups: security group to associate with each node of the cluster.
+                Typically used to provide access to external hosts
+
+            bastion_node_public_dns: Specify the Bastion host public DNS name if private subnet is provided.
+                Do not specify this value for a public subnet.
+
+            master_elastic_ip: It is the Elastic IP address for attaching to the cluster master
+
+        '''
+
+        self.set_compute_config(use_account_compute_creds, compute_access_key,
+                                compute_secret_key, role_instance_profile)
+        self.set_location(aws_region, aws_availability_zone)
+        self.set_network_config(bastion_node_public_dns, persistent_security_groups,
```

Comment:
Why don't we have self.set_storage_config in aws. Have we taken care of duplicating the compute config as storage config in the aws v2 payload?